numpyro/primitives.py

```diff
@@ -292,9 +316,9 @@ def deterministic(name, value):
     :param jnp.ndarray value: deterministic value to record in the trace.
     """
     if not _PYRO_STACK:
-        return value
+        return jnp.asarray(value)
```

Comment:
Could we return ArrayLike and avoid using asarray here and others? I use numpyro with string types so jnp.asarray does not work.

Reply:
Done! https://github.com/pyro-ppl/numpyro/pull/1940/commits/19e7cfc01fd37cb171d06a4990cddc9865c98fda